src/sol_binder/nonce/base.py

```diff
@@ -58,13 +65,53 @@ def sync_from_chain(self, new_accounts: List[HexAddress] = tuple()):
     def _sync_from_chain(self, account: HexAddress):
         return self._set(account, self.__w3.eth.get_transaction_count(account))
 
-    def get_and_increment(self, account: HexAddress) -> Nonce:
-        nonce = self._get_and_increment(account)
-        self._get_logger().debug(f"Nonce for {account}: {nonce}")
-        return nonce
+    def _lock(self):
+        raise NotImplementedError
 
-    def _get_and_increment(self, account: HexAddress) -> Nonce:
-        raise NotImplementedError()
+    def _unlock(self):
+        raise NotImplementedError
+
+    def _is_locked(self):
+        raise NotImplementedError
+
+    @contextmanager
+    def _lock_context(self):
+        now = time()
+        while time() - now < self._LOCK_TIMEOUT_SECONDS and self._is_locked():
```

Comment:
This is unsafe locking,
Locks should use the built-in locking library blocking-lock

I suggest renaming _lock to _blocking_lock to portray that the implementation should be blocking
If done right _is_locked can be completely discarded